src/IdealizedSimulations/neverworld_simulation.jl

```diff
@@ -0,0 +1,298 @@
+# Neverworld
+#
+# Ingredients:
+#
+#   * Zonally-periodic domain with continental shelves on all boundaries except Southern Ocean
+#       * longitude = (0, 60)
+#       * 2 configurations in latitude
+#           - Half basin: latitude = (-70, 0)
+#           - Full basin: latitude = (-70, 70)
+#       * z to -4000m
+#       * Southern Ocean channel from -60 to -40 (with a ridge to -2000 m)
+#
+#   * Zonally-homogeneous wind stress with mid-latitude jet and trade winds
+#
+#   * Buoyancy 
+#       * restoring hot at the equator and cold at the poles (parabola, cosine, smooth step function)
+#       * equator-pole buoyancy differential: 0.06 (α * g * 30 ≈ 0.06 with α=2e-4, g=9.81)
+#       * exponential initial vertical stratification with N² = 6e-5 and decay scale h = 1000 m
+#           - eg bᵢ = N² * h * exp(z / h)
+#
+#   * Quadratic bottom drag with drag_coefficient = 1e-3
+
+using CubicSplines
+
+using Oceananigans
+using Oceananigans.Units
+using Oceananigans.Grids: xnode, ynode
+using Oceananigans.Operators: Δzᶜᶜᶜ
+using Oceananigans.Coriolis: ActiveCellEnstrophyConservingScheme
+using Oceananigans.TurbulenceClosures: CATKEVerticalDiffusivity
+
+using ClimaOcean: CubicSplineFunction,
+                  u_bottom_drag, u_immersed_bottom_drag,
+                  v_bottom_drag, v_immersed_bottom_drag
+                  
+using ..VerticalGrids: stretched_vertical_cell_interfaces
+
+const c = Center()
+const f = Face()
+
+#####
+##### Geometry
+#####
+
+struct Point{T}
+    x :: T
+    y :: T
+end
+
+struct LineSegment{T}
+    p₁ :: Point{T}
+    p₂ :: Point{T}
+end
+
+struct Line{T}
+    p₁ :: Point{T}
+    p₂ :: Point{T}
+end
+
+distance(p₁::Point, p₂::Point) = sqrt((p₁.x - p₂.x)^2 + (p₁.y - p₂.y)^2)
+
+# https://stackoverflow.com/questions/849211/shortest-distance-between-a-point-and-a-line-segment
+function distance(point::Point, line::LineSegment)
+    x, y = point.x, point.y
+    x₁, y₁ = line.p₁.x, line.p₁.y
+    x₂, y₂ = line.p₂.x, line.p₂.y
+
+    # Line segment lengths
+    Δx = x₂ - x₁
+    Δy = y₂ - y₁
+
+    # Fractional increment to closest segment point
+    ϵ = ((x - x₁) * Δx + (y - y₁) * Δy) / (Δx^2 + Δy^2)
+    ϵ = clamp(ϵ, 0, 1)
+
+    # Closest segment point
+    x′ = x₁ + ϵ * Δx
+    y′ = y₁ + ϵ * Δy
+    
+    return sqrt((x′ - x)^2 + (y′ - y)^2)
+end
+
+function distance(point::Point, line::Line)
+    x, y = point.x, point.y
+    x₁, y₁ = line.p₁.x, line.p₁.y
+    x₂, y₂ = line.p₂.x, line.p₂.y
+    num = abs((x₂ - x₁) * (y₁ - y) - (y₂ - y₁) * (x₁ - x))
+    den = sqrt((x₂ - x₁)^2 + (y₂ - y₁)^2)
```

Comment:
this is correct, but the way it's written it's very cryptic for me :)
I'd prefer to see it in terms of dot products of vectors :)

```Julia
using LinearAlgebra

 𝐱 = [x, y, 0]
 𝐱₁ = [x₁, y₁, 0]
 𝐱₂ = [x₂, y₂, 0]

return abs(cross(𝐱₂ - 𝐱₁,  𝐱₁ -  𝐱)) / norm(𝐱₂ - 𝐱₁)
```

Reply:
why the third component?

Reply:
because `cross` doesn't work for 2D vectors :)